core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1456,16 +1461,25 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
         DBIO.successful(RequestComplete(StatusCodes.OK, SubmissionValidationReport(submissionRequest, header, succeeded, failed)))
     }
 
-  def getSubmissionStatus(workspaceName: WorkspaceName, submissionId: String) = {
-    dataSource.inTransaction { dataAccess =>
+  def getSubmissionStatus(workspaceName: WorkspaceName, submissionId: String): Future[PerRequestMessage] = {
+    val submissionWithoutCosts = dataSource.inTransaction { dataAccess =>
       withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, dataAccess) { workspaceContext =>
         withSubmission(workspaceContext, submissionId, dataAccess) { submission =>
           withUser(submission.submitter, dataAccess) { user =>
-            DBIO.successful(RequestComplete(StatusCodes.OK, new SubmissionStatusResponse(submission, user)))
+            DBIO.successful(submission, user)
           }
         }
       }
     }
+
+    submissionWithoutCosts flatMap {
+      case (submission, user) => {
+        val allWorkflowIds: Seq[String] = submission.workflows.flatMap(_.workflowId)
+        submissionCostService.getWorkflowCosts(workspaceName.namespace, allWorkflowIds, GoogleProject(workspaceName.namespace)) map { costMap =>
+          RequestComplete((StatusCodes.OK, new SubmissionStatusResponse(submission, costMap, user)))
+        }
```

Comment:
I think we should add some error-trapping here - at least we should be intentional about what happens in the case of error.

My preference is that if we can successfully retrieve the submission (from mysql) but encounter an error getting costs (from BQ), then we should return the submission without cost info - which needs error-trapping.

If you argue that a BQ error should totally prevent the submission response, then we don't need error trapping :)

Reply:
Done! Any point in writing to logs in the error case?